junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/InvocationInterceptor.java

```diff
@@ -218,14 +218,15 @@ default void interceptAfterAllMethod(Invocation<@Nullable Void> invocation,
 	 * @since 5.5
 	 */
 	@API(status = STABLE, since = "5.10")
-	interface Invocation<T extends @Nullable Object> {
+	interface Invocation<T> {
 
 		/**
 		 * Proceed with this invocation.
 		 *
 		 * @return the result of this invocation; potentially {@code null}.
 		 * @throws Throwable in case the invocation failed
 		 */
+		@Nullable
 		T proceed() throws Throwable;
```

Comment:
Does it really make sense to force all `Invocation` implementations to return `@Nullable` value?
Would it be better to keep an older `Invocation<T extends @Nullable Object>` approach so every implementation declares if it really wants to produce nullable value or not?